app/models/manageiq/providers/ansible_tower/configuration_manager/job.rb

```diff
@@ -34,6 +34,9 @@ def refresh_ems
     ext_management_system.with_provider_connection do |connection|
       update_with_provider_object(connection.api.jobs.find(ems_ref))
     end
+  rescue Faraday::ResourceNotFound
+    msg = "AnsibleTower Job #{name} with id(#{id}) does not exist on #{ext_management_system.name}"
+    raise MiqException::MiqOrchestrationStackNotExistError, msg
```

Comment:
@bdunne I think it is fine. `Job` is a subclass of `OrchestrationStack`. We could introduce a set of  error classes for Job, each subclassed from OrchestrationStack's corresponding error. Maybe too much.